File: 프로그래머스/0/120818. 옷가게 할인 받기/옷가게 할인 받기.swift
import Foundation

func solution(_ price: Int) -> Int {
    var discountPrice = 0
    
    // 할인 조건 적용
    if price >= 500_000 {
        discountPrice = price * 80 / 100  // 20% 할인
    } else if price >= 300_000 {
        discountPrice = price * 90 / 100  // 10% 할인
    } else if price >= 100_000 {
        discountPrice = price * 95 / 100  // 5% 할인
    } else {
        discountPrice = price  // 할인 없음
    }

    return discountPrice
}

File: 프로그래머스/0/120818. 옷가게 할인 받기/옷가게 할인 받기.js
// function solution(price) {
//     var answer = 0;
//     if (price < 100000) {
//         return price;
//     } else if (price >= 100000) {
//         return price * 0.95;
//     } else if (price >= 300000) {
//         return price * 0.90;
//     } else (price >= 500000) {
//         return price * 0.80
//     } 
// }

function solution(price) {
    if (price >= 500000) {
        return Math.trunc(price * 0.8);
    } else if (price >= 300000) {
        return Math.trunc(price * 0.9);
    } else if (price >= 100000) {
        return Math.trunc(price * 0.95);
    } else {
        return price
    }
}
